tests/utils/AABB3DTest.h

```diff
@@ -0,0 +1,58 @@
+//Copyright (c) 2017 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#ifndef TEMPLATE_TEST_H
+#define TEMPLATE_TEST_H
+
+#include <cppunit/TestFixture.h>
+#include <cppunit/extensions/HelperMacros.h>
+
+#include <../src/utils/AABB3D.h>
+
+namespace cura
+{
+
+class TemplateTest : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE(TemplateTest);
+    CPPUNIT_TEST(smokeTest);
+    CPPUNIT_TEST(hitTest);
+    CPPUNIT_TEST(hitTest2);
+    CPPUNIT_TEST(hitTest3);
+    CPPUNIT_TEST(hitTest4);
+    CPPUNIT_TEST(includeTest);
+    CPPUNIT_TEST(includeZTest);
+    CPPUNIT_TEST(offsetTest);
+    CPPUNIT_TEST(offsetTest2);
+    CPPUNIT_TEST(expandXYTest);
+    CPPUNIT_TEST_SUITE_END();
+
+public:
+    void setUp();
+
+    void tearDown();
+
+    void smokeTest();
+    void hitTest();
+    void hitTest2();
+    void hitTest3();
+    void hitTest4();
+    void includeTest();
+    void includeZTest();
+    void offsetTest();
+    void offsetTest2();
+    void expandXYTest();
+
+private:
+    /*!
+     * \brief The maximum allowed error in floats.
+     */
+    static constexpr double epsilon = 0.000001f;
```

Comment:
`coord_t` is an integer-type so an epsilon of 0.000001 wouldn't make sense. It just gets rounded down to 0. If it's changed to `coord_t` it has a different function.

In this particular case, `epsilon` is being used both to compare the values of transformation matrices (in which case 0.000001f makes sense) and to compare the values of (integer) coordinates in microns (in which case an epsilon of 10 or 15 makes more sense).